File: Scala/src/test/scala/FizzBuzzSpec.scala
import org.scalatest._

class FizzBuzzSpec extends FunSpec {
  describe("A FizzBuzz") {

    it("should return 'Fizz' from a multiple of three only") {
      assert(FizzBuzz.eval(3) == "Fizz")
      assert(FizzBuzz.eval(6) == "Fizz")
      assert(FizzBuzz.eval(9) == "Fizz")
    }
    it("should return 'Buzz' from a multiple of five only") {
      assert(FizzBuzz.eval(5) == "Buzz")
      assert(FizzBuzz.eval(10) == "Buzz")
    }
    it("should return 'FizzBuzz' from a multiple of three and five") {
      assert(FizzBuzz.eval(15) == "FizzBuzz")
      assert(FizzBuzz.eval(30) == "FizzBuzz")
    }
    it("should return the stringified input from a non multiple of three or five") {
      assert(FizzBuzz.eval(1) == "1")
      assert(FizzBuzz.eval(2) == "2")
    }
  }
}

File: Kotlin/src/test/kotlin/FizzBuzzSpec.kt
import org.jetbrains.spek.api.Spek
import org.jetbrains.spek.api.dsl.describe
import org.jetbrains.spek.api.dsl.it
import org.jetbrains.spek.api.dsl.on
import kotlin.test.assertEquals

object FizzBuzzSpec: Spek({
    describe("A FizzBuzz") {
        on("from a multiple of three only") {
            it("should return 'Fizz'") {
                assertEquals("Fizz", FizzBuzz.eval(3))
                assertEquals("Fizz", FizzBuzz.eval(6))
                assertEquals("Fizz", FizzBuzz.eval(9))
            }
        }
        on("from a multiple of five only") {
            it("should return 'Buzz'") {
                assertEquals("Buzz", FizzBuzz.eval(5))
                assertEquals("Buzz", FizzBuzz.eval(10))
            }
        }
        on("from a multiple of three and five") {
            it("should return 'FizzBuzz'") {
                assertEquals("FizzBuzz", FizzBuzz.eval(15))
                assertEquals("FizzBuzz", FizzBuzz.eval(30))
            }
        }
        on("from a non multiple of three or five") {
            it("should return the stringified input") {
                assertEquals("1", FizzBuzz.eval(1))
                assertEquals("2", FizzBuzz.eval(2))
                assertEquals("4", FizzBuzz.eval(4))
            }
        }
    }
})
